File: app/src/main/java/za/org/grassroot2/model/util/GroupPermissionChecker.java
package za.org.grassroot2.model.util;

import za.org.grassroot2.model.Group;
import za.org.grassroot2.model.GroupPermission;

/**
 * Created by qbasso on 20.10.2017.
 */

public class GroupPermissionChecker {

    public static boolean hasCreatePermission(Group g) {
        return g.getPermissions().contains(GroupPermission.CREATE_GROUP_MEETING) ||
                g.getPermissions().contains(GroupPermission.CREATE_GROUP_ENTRY) ||
                g.getPermissions().contains(GroupPermission.CREATE_GROUP_VOTE);
    }

    public static boolean canCallMeeting(Group g) {
        return g.getPermissions().contains(GroupPermission.CREATE_GROUP_MEETING);
    }

    public static boolean canCreateVote(Group g) {
        return g.getPermissions().contains(GroupPermission.CREATE_GROUP_VOTE);
    }

    public static boolean canCreateTodo(Group g) {
        return g.getPermissions().contains(GroupPermission.CREATE_GROUP_ENTRY);
    }
}

File: app/src/main/java/za/org/grassroot2/model/util/GroupPermissionChecker.kt
package za.org.grassroot2.model.util

import za.org.grassroot2.model.Group
import za.org.grassroot2.model.GroupPermission

/**
 * Created by qbasso on 20.10.2017.
 */

object GroupPermissionChecker {

    fun hasCreatePermission(g: Group): Boolean {
        return g.permissions.contains(GroupPermission.CREATE_GROUP_MEETING) ||
                g.permissions.contains(GroupPermission.CREATE_GROUP_ENTRY) ||
                g.permissions.contains(GroupPermission.CREATE_GROUP_VOTE)
    }

    fun canCallMeeting(g: Group?): Boolean {
        return g?.permissions?.contains(GroupPermission.CREATE_GROUP_MEETING) ?: false
    }

    fun canCreateVote(g: Group?): Boolean {
        return g?.permissions?.contains(GroupPermission.CREATE_GROUP_VOTE) ?: false
    }

    fun canCreateTodo(g: Group?): Boolean {
        return g?.permissions?.contains(GroupPermission.CREATE_GROUP_ENTRY) ?: false
    }
}
